test/unit/logic/transaction.js

```diff
@@ -0,0 +1,466 @@
+
+		it('should throw an error with no param', function () {
+			expect(transaction.create).to.throw();
+		});
+
+		it('should create a transaction', function () {
+			var trsData = getValidTransactionData();
+			expect(transaction.create(trsData)).to.be.an('object');
+		});
+
+		it('should throw an error when sender is not set', function () {
+			var trsData = getValidTransactionData();
+			delete trsData.sender;
+			expect(transaction.create.bind(transaction, trsData)).to.throw();
+		});
+
+		it('should throw an error when keypair is not set', function () {
+			var trsData = getValidTransactionData();
+			delete trsData.keypair;
+			expect(transaction.create.bind(transaction, trsData)).to.throw();
+		});
+
+		it('should return transaction with optional data field', function () {
+			var trsData = getValidTransactionData();
+			expect(transaction.create(trsData).data).to.be.a('string');
+		});
+
+	});
+
+	describe('attachAssetType', function () {
+
+		it('should attach all transaction types', function () {
+			attachAllAssets(transaction);
+		});
+
+
+		it('should throw an error with no param', function () {
+			expect(transaction.attachAssetType).to.throw();
+		});
+	});
+
+	describe('sign', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.sign).to.throw();
+		});
+
+		it('should sign transaction', function () {
+			var trsData = getValidTransactionData();
+			var createdTransaction = transaction.create(trsData);
+			expect(transaction.sign(trsData.keypair, createdTransaction)).to.be.ok;
+		});
+
+		it('should update signature when data is changed', function () {
+			var trsData = getValidTransactionData();
+			var createdTransaction = transaction.create(trsData);
+			var trsSignature = transaction.sign(trsData.keypair, createdTransaction);
+			createdTransaction.data = 'different data';
+			var updatedTrsSignature = transaction.sign(trsData.keypair, createdTransaction);
+			expect(trsSignature).to.not.equal(updatedTrsSignature);;
+
+		});
+	});
+
+	describe('multisign', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.multisign).to.throw();
+		});
+	});
+
+	describe('getId', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.getId).to.throw();
+		});
+	});
+
+	describe('getHash', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.getHash).to.throw();
+		});
+	});
+
+	describe('getBytes', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.getBytes).to.throw();
+		});
+
+		it('should return same result when called multiple times', function () {
+			attachAllAssets(transaction);
+			var firstCalculation = transaction.getBytes(validTransaction);
+			var secondCalculation = transaction.getBytes(validTransaction);
+			expect(firstCalculation.equals(secondCalculation)).to.be.ok;
+		});
+
+	});
```

Comment:
Here we should check byte calculation works with and without the optional data property.